pkg/controller/elasticsearch/name/name.go

```diff
@@ -29,10 +31,62 @@ const (
 	transportCertificatesSecretSuffix = "transport-certificates"
 )
 
-// ESNamer is a Namer that is configured with the defaults for resources related to an ES cluster.
-var ESNamer = name.Namer{
-	MaxSuffixLength: MaxSuffixLength,
-	DefaultSuffixes: []string{"es"},
+var (
+	// ESNamer is a Namer that is configured with the defaults for resources related to an ES cluster.
+	ESNamer = common_name.NewNamer("es")
+
+	suffixes = []string{
+		configSecretSuffix,
+		secureSettingsSecretSuffix,
+		httpServiceSuffix,
+		elasticUserSecretSuffix,
+		xpackFileRealmSecretSuffix,
+		internalUsersSecretSuffix,
+		unicastHostsConfigMapSuffix,
+		licenseSecretSuffix,
+		defaultPodDisruptionBudget,
+		scriptsConfigMapSuffix,
+		transportCertificatesSecretSuffix,
+	}
+)
+
+// Validate checks the validity of resource names that will be generated by the given Elasticsearch object.
+func Validate(es v1alpha1.Elasticsearch) error {
+	esName := k8s.ExtractNamespacedName(&es).Name
+	if len(esName) > common_name.MaxResourceNameLength {
+		return fmt.Errorf("name exceeds maximum allowed length of %d", common_name.MaxResourceNameLength)
+	}
+
+	// validate ssets
+	for _, nodeSpec := range es.Spec.Nodes {
+		if errs := apimachineryvalidation.NameIsDNSSubdomain(nodeSpec.Name, false); len(errs) > 0 {
+			return fmt.Errorf("invalid nodeSpec name '%s': [%s]", nodeSpec.Name, strings.Join(errs, ","))
+		}
+
+		ssetName, err := ESNamer.SafeSuffix(esName, nodeSpec.Name)
+		if err != nil {
+			return errors.Wrapf(err, "error generating StatefulSet name for nodeSpec: '%s'", nodeSpec.Name)
+		}
+
+		// length of the ordinal suffix that will be added to the pods of this sset
+		podOrdinalSuffixLen := len(strconv.FormatInt(int64(nodeSpec.NodeCount), 10))
```

Comment:
nit: should this be `nodeSpec.NodeCount - 1`? We start numbering from 0, so for 100 nodes the last ordinal will be 99.

Reply:
It was intentional to account for the dash before the ordinal but now that I think about it, it doesn't work for node counts below 10.